src/MKL.jl

```diff
@@ -16,41 +16,39 @@ end
     THREADING_GNU
     THREADING_TBB
 end
+
 @enum Interface begin
     INTERFACE_LP64
     INTERFACE_ILP64
     INTERFACE_GNU
 end
 
 function set_threading_layer(layer::Threading = THREADING_INTEL)
-    err = ccall((:MKL_Set_Threading_Layer, libmkl_rt), Cint,
-        (Cint,), layer)
-    if err == -1
-        throw(ErrorException("return value was -1"))
-    end
+    err = ccall((:MKL_Set_Threading_Layer, libmkl_rt), Cint, (Cint,), layer)
+    err == -1 && throw(ErrorException("return value was -1"))
     return nothing
 end
 
-function set_interface_layer(interface = BlasInt == Int64 ? INTERFACE_ILP64 : INTERFACE_LP64)
-    err = ccall((:MKL_Set_Interface_Layer, libmkl_rt), Cint,
-        (Cint,), interface)
-    if err == -1
-        throw(ErrorException("return value was -1"))
-    end
+function set_interface_layer(interface = Base.USE_BLAS64 ? INTERFACE_ILP64 : INTERFACE_LP64)
+    err = ccall((:MKL_Set_Interface_Layer, libmkl_rt), Cint, (Cint,), interface)
+    err == -1 && throw(ErrorException("return value was -1"))
     return nothing
 end
 
 function __init__()
-    set_threading_layer()
-    set_interface_layer()
+    if MKL_jll.is_available()
+        set_threading_layer()
+        set_interface_layer()
+        VERSION > v"1.7.0-DEV.623" && BLAS.lbt_forward(libmkl_rt, clear=true)
+    end
 end
 
-mklnorm(x::Vector{Float64}) = ccall((:dnrm2_, libmkl_rt), Float64,
-    (Ref{BlasInt}, Ptr{Float64}, Ref{BlasInt}),
-    length(x), x, 1)
-
+function mklnorm(x::Vector{Float64})
+    ccall((:dnrm2_, libmkl_rt), Float64,
+          (Ref{MKLBlasInt}, Ptr{Float64}, Ref{MKLBlasInt}),
+          length(x), x, 1)
+end
 
-# Include installation routines as well
-include("install.jl")
+VERSION > v"1.7.0-DEV.623" && include("install.jl")
```

Comment:
Isn't this the wrong direction? We want to `include` the install file if we do not have access to LBT.

Reply:
That is what the code does.

https://github.com/JuliaLinearAlgebra/MKL.jl/blob/259493b6e811c60fc2436b8efdd65665a91c5bd7/src/MKL.jl#L54

Reply:
This is on 1.6:

```
julia> JULIA_VER_NEEDED = v"1.7.0-DEV.641"
v"1.7.0-DEV.641"

julia> VERSION > JULIA_VER_NEEDED
false
```

so the LHS is false and the `include` is not executed on 1.6. And on 1.6 I cannot execute this:

```
julia> VERSION
v"1.6.2"

julia> using MKL

julia> MKL.enable_mkl_startup()
ERROR: UndefVarError: enable_mkl_startup not defined
```

However, on 1.7 it is defined:

```
julia> MKL.enable_mkl_startup()
[ Info: Splicing in code to load MKL in /Users/kristoffercarlsson/julia/usr/bin/../share/julia/base/sysimg.jl
ERROR: UndefVarError: USEBLAS64 not defined
```

So I think the conditional should be reversed here.